File: C++/I/05-FORMS/2/form2.cpp
#include "form2.h"
#include "ui_form2.h"

Form2::Form2(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::Form2)
{
    ui->setupUi(this);
}

Form2::~Form2()
{
    delete ui;
}

void Form2::closeEvent(QCloseEvent *)
{
    setVisible(false);
    emit textChanged();
}

void Form2::changeVisible()
{
    setVisible(!isVisible());
    emit textChanged();
}

File: Python/I/05-FORMS/2/form2.py
from ui_form2 import *
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *

class Form2(Ui_Form2, QDialog):
    visibleChanged = pyqtSignal(bool)
    
    def __init__(self, parent):
        super(self.__class__, self).__init__(parent)
        self.setupUi(self)

    def showEvent(self, event):
        self.visibleChanged.emit(False)

    def closeEvent(self, event):
        self.visibleChanged.emit(True)
